### ISSUE TITLE
iotedge running before, now it doesn't run because disk space full

### ISSUE THREAD

**Issue by gadgetman4u** [Azure/azure-iotedge#18]
Hi,

I had my iotedge running before with my deployed modules on a Raspberry Pi 3 (running Raspbian), but then over the weekend, the iotedge stopped running.  When I type sudo iotedge list, I get:

An error in the management http client occurred.
caused by: Hyper error
caused by: an error occurred trying to connect: Connection refused (os error 111)

I also see a black screen when I rebooted the Raspberry Pi 3 and see no GUI showing up with the login screen.  I suspected that I ran out of disk space on the Raspberry Pi 3 and it looks like I did.  I found out that Docker is taking the most of my space, so I tried typing:

sudo docker system prune -a -f

It worked before, but now it gives this error:

Cannot connect to the Docker daemon at unix://var/run/docker.sock.  Is the docker daemon running?

I am not sure what else to do.  Plus, I'm running tensorflow on one of my iotedge modules, does tensorflow take up so much space in Docker that it uses up all my disk space on my SD card (32 G) on the Raspberry Pi 3?

Thanks for any help.

**Comment by veyalla** [comment]
The log files might be taking up too much space. Guidance below might help:
https://docs.microsoft.com/en-us/azure/iot-edge/production-checklist#solution-management

**Comment by gadgetman4u** [comment]
@veyalla thanks.  How do I use the logrotate tool?  Do I need to create a json log file and then put those commands in there?  Also, there doesn't seem to be an /etc/docker/daemon.json file so I cannot make changes to the log size.

**Comment by gadgetman4u** [comment]
@veyalla On closer examination, it seems like the /var/lib/docker/overlay2 directory is taking up 29G of space.  On other sites, it seems that others have issues where the overlay2 keeps on growing in size: 

https://forums.docker.com/t/some-way-to-clean-up-identify-contents-of-var-lib-docker-overlay/30604/21

**Comment by gadgetman4u** [comment]
@veyalla  I was able to clean up the Docker overlay directory using:

https://forums.docker.com/t/some-way-to-clean-up-identify-contents-of-var-lib-docker-overlay/30604/19

which cleared up 21G of space!  I then followed:

https://blog.boatswain.io/post/docker-container-log-rotation/

to setup the log rotation.  I set it to 1 G but I see now that the overlay is using 15G out of 30G and now has 14G remaining.  

How to I eliminate the logs to be written to the device (or at least limit to just 1 G and not more) and just only sent over to the Azure LogAnalytics using your log IoT edge module?

**Comment by veyalla** [comment]
Hmm, will need to research this. 

Can you share the output of “docker version” from the system?

**Comment by gadgetman4u** [comment]
@veyalla Output from "docker version":

Client:
 Version:      18.06.0-dev
 API version:  1.37
 Go version:   go1.10.2
 Git commit:   daf021fe
 Built:        Wed Jun 27 00:53:52 2018
 OS/Arch:      linux/arm
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      dev
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.10.2
  Git commit:   9aac187
  Built:        Tue Jun 26 23:27:27 2018
  OS/Arch:      linux/arm
  Experimental: false

**Comment by veyalla** [comment]
It would be good to check if you’re seeing the behavior with the official Moby/Docker runtime for IoT Edge:

https://docs.microsoft.com/en-us/azure/iot-edge/how-to-install-iot-edge-linux-arm#install-the-container-runtime

**Comment by gadgetman4u** [comment]
@veyalla So I should install the Moby container runtime? I thought I had already done this when installing the Azure IoTedge for Raspberry Pi.

**Comment by veyalla** [comment]
Yes, the container runtime is a separate install as documented here: https://docs.microsoft.com/en-us/azure/iot-edge/how-to-install-iot-edge-linux-arm

**Comment by gadgetman4u** [comment]
@veyalla ok, I have done that and this is the result of "docker version":

Client:
 Version:           3.0.2
 API version:       1.40
 Go version:        go1.11.1
 Git commit:        e042b58f
 Built:             Tue Dec  4 17:16:17 2018
 OS/Arch:           linux/arm
 Experimental:      false

Server:
 Engine:
  Version:          3.0.2
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.11.2
  Git commit:       74baf62
  Built:            Tue Dec  4 23:30:17 2018
  OS/Arch:          linux/arm
  Experimental:     false

**Comment by gadgetman4u** [comment]
@veyalla another question, why does the edgeHub fail when other IoT modules are running? Seems I have to wait a couple of minutes before it becomes "running" or just restart the iotedge.

**Comment by veyalla** [comment]
We'll need to look into the logs to figure it out. I suggest you open an issue in https://github.com/azure/iotedge/issues. This repo mainly houses the product binaries built out of the iotedge open source project.
